tests/webapp/api/test_push_api.py

```diff
@@ -124,26 +124,26 @@ def test_resultset_list_single_long_revision(webapp, eleven_jobs_stored, test_re
     )
 
 
-def test_resultset_list_single_long_revision_stored_long(webapp, sample_resultset, test_repository):
+def test_push_list_single_long_revision_stored_long(webapp, sample_push, test_repository):
     """
-    test retrieving a resultset list with store long revision, filtered by a single long revision
+    test retrieving a push list with store long revision, filtered by a single long revision
     """
     long_revision = "21fb3eed1b5f3456789012345678901234567890"
 
-    # store a resultset with long revision
-    resultset = copy.deepcopy(sample_resultset[0])
-    resultset["revisions"][0]["revision"] = long_revision
-    store_result_set_data(test_repository, [resultset])
+    # store a push with long revision
+    push = copy.deepcopy(sample_push[0])
+    push["revisions"][0]["revision"] = long_revision
+    store_push_data(test_repository, [push])
 
     resp = webapp.get(
-        reverse("resultset-list", kwargs={"project": test_repository.name}),
+        reverse("push-list", kwargs={"project": test_repository.name}),
         {"revision": long_revision}
     )
     assert resp.status_int == 200
     results = resp.json['results']
     meta = resp.json['meta']
     assert len(results) == 1
-    assert set([rs["revision"] for rs in results]) == {sample_resultset[0]['revision']}
+    assert set([rs["revision"] for rs in results]) == {sample_push[0]['revision']}
```

Comment:
ahh, yeah.  good call.  I hadn't noticed that about the naming.

So this type of look is called "list comprehension"  It's a really cool and powerful part of python: http://treyhunner.com/2015/12/python-list-comprehensions-now-in-color/

In this case, the loop is saying: Create a new list of items that are just the ``revision`` field from each item in the ``results`` list.  And the ``set`` part around it takes that new list and make it into a ``set`` object instead.  A ``list`` can have duplicates, but a ``set`` can't.  So making it into a ``set`` will automatically remove all dupes.  The cost is that it will lose the order because ``sets`` don't have order.  But in this case, we don't care about order.